daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -1014,36 +1009,30 @@ case class ParserTestCase(ptc: NodeSeq, parentArg: DFDLTestSuite)
 
     val testDataLength = lengthLimitInBits
 
-    val optExtVarDiag =
-      try {
-        processor = processor.withExternalDFDLVariables(externalVarBindings)
-        None
-      } catch {
-        case diag: Diagnostic =>
-          Some(diag)
-      }
-    if (optExtVarDiag.isDefined) {
-      toss(optExtVarDiag.get, implString)
-    } else {
-      val actual = processor.parse(new ByteArrayInputStream(testData), testDataLength)
-      val diagObjs = actual.getDiagnostics
+    try {
+      processor = processor.withExternalDFDLVariables(externalVarBindings)
+    } catch {
+      case e: Exception => toss(e, implString)
```

Comment:
Is this toss just for a centralized breakpoint/print location?
If that's the case, why not catch Throwable?
If not, then this should be narrower set of exception classes.

Reply:
`withExternalDFDLVairables` should only throw an ExternalVariableException, so the toss just converts that exception into a TDMLRunnerException to make things a little nicer looking. This should probably catch `ExternalVariableException` instead of `Exception`

Reply:
Actually, the ExternalVariableException is defined in daffodil-runtime1. Which the TDMLRUnner doesn't have access to. In fact, it looks like all of the TDMLRunner only catches Exception or Throwable, probably for this reason. So I think Exception here is probably the right thing, with the understanding that the only Exception this should throw is the variable ones. And even if it's not, we're still going to throw a TDMLException and the test will fail, so doesn't really change the behavior.